signature/types.go

```diff
@@ -1,103 +1,149 @@
 package signature
 
-import "crypto"
+import (
+	"crypto/x509"
+	"errors"
+	"time"
+)
 
 // SignatureMediaType list the supported media-type for signatures.
 type SignatureMediaType string
 
-// SignatureAlgorithm lists supported signature algorithms.
-type SignatureAlgorithm string
-
-// HashAlgorithm algorithm associated with the key spec.
-type HashAlgorithm string
+// SigningScheme formalizes the feature set (guarantees) provided by
+// the signature.
+// Reference: https://github.com/notaryproject/notaryproject/blob/main/signing-scheme.md
+type SigningScheme string
 
-// One of following supported specs
-// https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#algorithm-selection
+// SigningSchemes supported by notation.
 const (
-	RSASSA_PSS_SHA_256 SignatureAlgorithm = "RSASSA_PSS_SHA_256"
-	RSASSA_PSS_SHA_384 SignatureAlgorithm = "RSASSA_PSS_SHA_384"
-	RSASSA_PSS_SHA_512 SignatureAlgorithm = "RSASSA_PSS_SHA_512"
-	ECDSA_SHA_256      SignatureAlgorithm = "ECDSA_SHA_256"
-	ECDSA_SHA_384      SignatureAlgorithm = "ECDSA_SHA_384"
-	ECDSA_SHA_512      SignatureAlgorithm = "ECDSA_SHA_512"
-)
+	// notary.x509 signing scheme.
+	SigningSchemeX509 SigningScheme = "notary.x509"
 
-// One of following supported specs
-// https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#algorithm-selection
-const (
-	SHA_256 HashAlgorithm = "SHA_256"
-	SHA_384 HashAlgorithm = "SHA_384"
-	SHA_512 HashAlgorithm = "SHA_512"
+	// notary.x509.signingAuthority schema.
+	SigningSchemeX509SigningAuthority SigningScheme = "notary.x509.signingAuthority"
 )
 
-// HashFunc returns the Hash associated k.
-func (h HashAlgorithm) HashFunc() crypto.Hash {
-	switch h {
-	case SHA_256:
-		return crypto.SHA256
-	case SHA_384:
-		return crypto.SHA384
-	case SHA_512:
-		return crypto.SHA512
-	}
-	return 0
+// MediaTypePayloadV1 is the supported content type for signature's payload.
+const MediaTypePayloadV1 = "application/vnd.cncf.notary.payload.v1+json"
```

Comment:
Given that in future we can have multiple mediatype. Does this deserves its own type?